features/graph/api/src/main/java/org/opennms/netmgt/graph/api/search/SearchSuggestion.java

```diff
@@ -28,21 +28,26 @@
 
 package org.opennms.netmgt.graph.api.search;
 
+import java.util.Comparator;
 import java.util.Objects;
 
+import com.google.common.base.MoreObjects;
+
 /**
  * A {@link SearchSuggestion} is provided to the user and was created from a (partial) search query.
  * A suggestion is a very abstract and may not directly related to a vertex. For example a category.
  * The main idea is, that a {@link SearchSuggestion} represents an item a user can select, which
  * afterwards is resolved to a List of vertices/edges.
  */
-public class SearchSuggestion {
+public class SearchSuggestion implements Comparable<SearchSuggestion> {
 
     // The context of the suggestion, e.g. category, attribute, node, etc. to allow for a more fine grain suggestion
     // This may be achieved later by sub-classing
     private String context;
     // The user-friendly label for the suggestion, which is shown to the user
     private String label;
+    // TODO: Patrick discuss with mvr: maybe we will need later also a technical identifier (instead of just a user
```

Comment:
Discuss with @mvrueden

Reply:
The `context` defines what type or what the search suggestion represents, e.g. a category, node, ip address, etc. 
The `label` is what is shown to the user, e.g. `My super Node`
The `provider` is required for resolution of the suggestion later on. It is just the `providerId`.

However, you are of course right, we probably need somekind of `internal identifier` in order to later on know, that this represents the node with id `5`.

Reply:
we decided to stick for now with the label but will keep the technical identifier in mind

Reply:
we decided to stick for now with the label but will keep the technical identifier in mind.